packages/router/utils.ts

```diff
@@ -441,19 +441,43 @@ function matchRouteBranch<
   return matches;
 }
 
+/**
+ * @private
+ * Parameters in the path
+ */
+ type PathParams<
+ Path extends string
+> = Path extends `:${infer Param}/${infer Rest}`
+   ? Param | PathParams<Rest>
+   : Path extends `:${infer Param}`
+     ? Param
+     : Path extends `${any}:${infer Param}`
+         ? PathParams<`:${Param}`>
+         : Path extends `${any}/*`
+           ? "*"
+           : Path extends "*"
+             ? "*" 
+             : never
```

Comment:
I'll preface this with the fact that I'm no TS expert :).  But It feels like this is doing something very similar (if not the same thing) as `ParamParseSegment`/`Params` higher up in this file?  Do you think there's any chance to leverage a shared type?

Also going to ping @pcattori on this one as our in-house TS guru

Reply:
To my eyes, this looks like a simpler and more complete version of `ParamParseSegment` (more complete, because it accounts for `"*"` params).

It also returns either a union of `string` or just `never` when parsing fails (instead of `ParamParseSegment`'s `type ParamParseFailed = { failed: true }`) , so its suitable to use directly as a property key.

If we can migrate or rewrite the comments from `ParamParseSegment` to `PathParams`, I'd be in favor of replacing `ParamParseSegment` with this.

Reply:
Naming-wise, I think singular nouns work better for unions. So `PathParam` feels better to me. Like "a path param can be any of these". But not a strong opinion there.

Reply:
>I think singular nouns work better for unions

I totally agree. To be honest, I don't have an opinion on the naming. So, this opinion looks good to me.

Reply:
@pcattori 
I did all tasks.
Please review my feature.